### DISCUSSION TITLE
Public Inquiry for Data - All Stations

### DISCUSSION THREAD

**Discussion by krowvin** [USACE/cwms-data-api#660] [General Aesthetics]
## The following was sent to us via Email:
*I am posting here for all to see and contribute on. Pending answers we can provide a link to this thread*

# Question

I am looking to ingest time series data for all water bodies which have water level and/or flow data in the CWMS data base. 

I’ve been digging through the [Swagger Documentation](https://cwms-data.usace.army.mil/cwms-data/swagger-ui.html), but I am still struggling to find the specific API call(s) that I need. 

## Is someone able to identify which API endpoint(s) I need to achieve the following?
* All stations which are operational which collect time series for water levels in rivers, coasts, reservoirs, etc.
    * I have identified the “locations” end point, but there am struggling to filter to only operational locations, and to find a quick way to identify if my time series of interest is available for any one station
* All stations which are operations which collect flow time series data for rivers
* The API endpoint to collect the most granular time series available for one or both of the above.

**Comment by Enovotny** [comment]
right now that would be to grab all of the timeseries from the timeseries catalog endpoint. The filter them by time series that have an extent.  Finally request data from them individually.  

Other option would be to grab all time series from the DMZ include list time series group.  grab the timeseries values from each individually.  

We are going to implement the time series catalog into cwms-python that the default would filter by time series that have an extent.  removing non public data.  We are also looking to open up the A2W designations table in CDA which would also make this easier to grab the "official" time series.  however both of these are not implemented at the moment so the above are the best bet.

**Comment by JeremyDKellett** [comment]
A2W (legacy and presumably its replacement) has a end point for all A2W locations (which answers the first question.

A2W also has a endpoint for identifying the cwms TS ID of those classifications

**Comment by MikeNeilson** [comment]
This has come up previously, an issue was opened but apparently we just forgot about it: https://github.com/USACE/cwms-data-api/issues/360

I think as part of both that and #620, another specific "show-official-only" type marker would be good, that could lookup with the A2W tables what's assigned for the displays.

I'm going to move this to discussion, since the behavior is already covered under two issues.

**Comment by JeremyDKellett** [comment]
A simple way to solve this, with the existing CDA structures  is to create a location category "Data Dissemination" and group "A2W" (or akin) and insert the associations at T7s or CPC/DMZ/CWBI. Likewise create a time series group to do the same for TS Codes.

The latter is where the data model may have challenges as one could have multiple ts_codes for each "Elev/Stage/Inflow/Outflow/TW stage/TW elev/Precip/etc." classification. But it would at least be discoverable.
